lib/Runtime/Library/JavascriptFunction.cpp

```diff
@@ -2793,6 +2793,10 @@ void __cdecl _alloca_probe_16()
             {
                 *value = scriptContext->GetLibrary()->GetNull();
             }
+            else
```

Comment:
funcCaller->GetScriptContext() != requestContext then in case of cross domain check fail we are returning null from scriptContext (not marshaled to requestContext). That's why I say in if (S_OK != hr) case use *value = nullValue which is from requestContext.